src/api/app/helpers/webui/request_helper.rb

```diff
@@ -174,4 +174,23 @@ def request_type_of_action(bs_request)
 
     bs_request.bs_request_actions.first.type.titleize
   end
+
+  def request_with_broken_build(buildresults)
+    buildresults.any? { |b| b != 'succeeded' && b != 'disabled' && b != 'excluded' && b != 'building' }
+  end
+
+  def quick_build_results_status(buildresults)
+    return unless request_with_broken_build(buildresults)
+
+    'fa-exclamation-triangle repository-state-broken'
+  end
+
+  def build_status_icon_from_action(action)
+    source_project = Project.find_by_name(action.source_project)
```

Comment:
I'm not sure if it must but this doesn't take any [links](https://github.com/openSUSE/open-build-service/wiki/Links) into account...

Reply:
Let me think about that

Reply:
As far as I could test, linked projects work perfectly fine

Reply:
Which can only mean  you did not test hard enough...